pkgs/development/libraries/nfft/default.nix

```
@@ -0,0 +1,42 @@
+{ autoconf
+, automake
+, cunit
+, fetchFromGitHub
+, fftw
+, lib
+, libtool
+, llvmPackages
+, stdenv
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "nfft";
+  version = "3.5.3";
+
+  src = fetchFromGitHub {
+    owner = "NFFT";
+    repo = "nfft";
+    rev = finalAttrs.version;
+    sha256 = "sha256-HR8ME9PVC+RAv1GIgV2vK6eLU8Wk28+rSzbutThBv3w=";
+  };
+
+  preConfigure = ''
+    bash bootstrap.sh
+  '';
+  nativeBuildInputs = [ autoconf automake cunit libtool ];
+  configureFlags = [
+    "--enable-all"
+    "--enable-openmp"
+    "--enable-portable-binary"
+  ];
+  buildInputs = lib.optionals stdenv.cc.isClang [ llvmPackages.openmp ];
+  propagatedBuildInputs = [ fftw ];
+  doCheck = true;
```

Nit: Could space these out
```suggestion
  preConfigure = ''
    bash bootstrap.sh
  '';
  
  nativeBuildInputs = [
    autoconf
    automake
    cunit
    libtool
  ];
  
  configureFlags = [
    "--enable-all"
    "--enable-openmp"
    "--enable-portable-binary"
  ];
  
  buildInputs = lib.optionals stdenv.cc.isClang [ llvmPackages.openmp ];
  
  propagatedBuildInputs = [ fftw ];
  
  doCheck = true;
```